pandas/core/indexes/period.py

```diff
@@ -219,29 +216,9 @@ def __new__(
 
         if data is None and ordinal is None:
             # range-based.
-            data, freq2 = PeriodArray._generate_range(start, end, periods, freq, fields)
-            # PeriodArray._generate range does validate that fields is
+            data, freq2 = PeriodArray._generate_range(None, None, None, freq, fields)
```

Comment:
does this actually work with None values?

Reply:
Yes, PeriodArray._generate_range dispatches to either _get_ordinal_range or _range_from_fields